server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -420,7 +420,7 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
     }
 
     protected final void doXContentAnalyzers(XContentBuilder builder, boolean includeDefaults) throws IOException {
-        if (fieldType.tokenized() == false) {
+        if (fieldType.tokenized() == false || fieldType().getTextSearchInfo().getSearchAnalyzer() == null) {
```

Comment:
Will this change the behavior when `includeDefaults` is `true` and there isn't a `search_analyzer`? I'm not familiar enough with the default values of everything to know. I think it is a least worth a comment about it here.

Reply:
It shouldn't do, because `doXContentAnalyzers` is only called from text-based field mappers, which ought to always have analyzers configured (if nothing is specified in config, then we'll end up with the default).  This should probably be an assertion again though.

Reply:
Turns out this is incorrect - it can return `null` here, in which case we need to use the default.  Which is all a bit of a pain, but will be cleaned up when we move mapper definitions to the Parametrized system.